scripts/lib/CIME/preview_namelists.py

```diff
@@ -70,8 +72,20 @@ def create_namelists(case, component=None):
         config_dir = os.path.dirname(config_file)
         if model_str == "cpl":
             compname = "drv"
+            complist = [m for m in models if m.upper() != "CPL"]
+            if cpl_ninst > 1:
```

Comment:
Please document what the xmlfact dictionary does and why its initialized the way it is. Similarly for xmlsave. Why are these two dictionaries needed here.